### PR TITLE
Api framework graphql interfaces mark

### LINKED ISSUE #32028: Mark Framework GraphQL interfaces as API
The following interfaces should be marked as API in 2.4-develop branch:

1. lib/internal/Magento/Framework/GraphQl/Config/ConfigElementFactoryInterface.php
1. lib/internal/Magento/Framework/GraphQl/Config/ConfigElementInterface.php
1. lib/internal/Magento/Framework/GraphQl/Config/Element/FieldInterface.php
1. lib/internal/Magento/Framework/GraphQl/Config/Element/OutputFieldInterface.php
1. lib/internal/Magento/Framework/GraphQl/Config/Element/TypeInterface.php
1. lib/internal/Magento/Framework/GraphQl/Config/Element/UnionInterface.php
1. lib/internal/Magento/Framework/GraphQl/ConfigInterface.php
1. lib/internal/Magento/Framework/GraphQl/Query/ErrorHandlerInterface.php
1. lib/internal/Magento/Framework/GraphQl/Query/Resolver/Argument/FieldEntityAttributesInterface.php
1. lib/internal/Magento/Framework/GraphQl/Query/Resolver/Argument/SearchCriteria/ArgumentApplierInterface.php
1. lib/internal/Magento/Framework/GraphQl/Query/Resolver/BatchRequestItemInterface.php
1. lib/internal/Magento/Framework/GraphQl/Query/Resolver/BatchResolverInterface.php
1. lib/internal/Magento/Framework/GraphQl/Query/Resolver/BatchServiceContractResolverInterface.php
1. lib/internal/Magento/Framework/GraphQl/Query/Resolver/ContextInterface.php
1. lib/internal/Magento/Framework/GraphQl/Query/Resolver/IdentityInterface.php
1. lib/internal/Magento/Framework/GraphQl/Query/Resolver/ResolveRequestInterface.php
1. lib/internal/Magento/Framework/GraphQl/Query/Resolver/TypeResolverInterface.php
1. lib/internal/Magento/Framework/GraphQl/Query/ResolverInterface.php
1. lib/internal/Magento/Framework/GraphQl/Schema/SchemaGeneratorInterface.php
1. lib/internal/Magento/Framework/GraphQl/Schema/Type/Entity/MapperInterface.php
1. lib/internal/Magento/Framework/GraphQl/Schema/Type/Enum/DataMapperInterface.php
1. lib/internal/Magento/Framework/GraphQl/Schema/Type/InputTypeInterface.php
1. lib/internal/Magento/Framework/GraphQl/Schema/Type/Output/ElementMapper/FormatterInterface.php
1. lib/internal/Magento/Framework/GraphQl/Schema/Type/OutputTypeInterface.php
1. lib/internal/Magento/Framework/GraphQl/Schema/Type/WrappedTypeInterface.php
1. lib/internal/Magento/Framework/GraphQl/Schema/TypeInterface.php
1. lib/internal/Magento/Framework/GraphQlSchemaStitching/GraphQlReader/TypeMetaReaderInterface.php

### PR DESCRIPTION
**Pull request by mrtuvn**
### Description (*)

Mark GraphQL interfaces as API

lib/internal/Magento/Framework/GraphQl/Config/ConfigElementFactoryInterface.php
lib/internal/Magento/Framework/GraphQl/Config/ConfigElementInterface.php
lib/internal/Magento/Framework/GraphQl/Config/Element/FieldInterface.php
lib/internal/Magento/Framework/GraphQl/Config/Element/OutputFieldInterface.php
lib/internal/Magento/Framework/GraphQl/Config/Element/TypeInterface.php
lib/internal/Magento/Framework/GraphQl/Config/Element/UnionInterface.php
lib/internal/Magento/Framework/GraphQl/ConfigInterface.php
lib/internal/Magento/Framework/GraphQl/Query/ErrorHandlerInterface.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/Argument/FieldEntityAttributesInterface.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/Argument/SearchCriteria/ArgumentApplierInterface.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/BatchRequestItemInterface.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/BatchResolverInterface.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/BatchServiceContractResolverInterface.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/ContextInterface.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/IdentityInterface.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/ResolveRequestInterface.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/TypeResolverInterface.php
lib/internal/Magento/Framework/GraphQl/Query/ResolverInterface.php
lib/internal/Magento/Framework/GraphQl/Schema/SchemaGeneratorInterface.php
lib/internal/Magento/Framework/GraphQl/Schema/Type/Entity/MapperInterface.php
lib/internal/Magento/Framework/GraphQl/Schema/Type/Enum/DataMapperInterface.php
lib/internal/Magento/Framework/GraphQl/Schema/Type/InputTypeInterface.php
lib/internal/Magento/Framework/GraphQl/Schema/Type/Output/ElementMapper/FormatterInterface.php
lib/internal/Magento/Framework/GraphQl/Schema/Type/OutputTypeInterface.php
lib/internal/Magento/Framework/GraphQl/Schema/Type/WrappedTypeInterface.php
lib/internal/Magento/Framework/GraphQl/Schema/TypeInterface.php
lib/internal/Magento/Framework/GraphQlSchemaStitching/GraphQlReader/TypeMetaReaderInterface.php

Reference classes/interfaces also marked api
lib/internal/Magento/Magento/Framework/GraphQl/Schema/Type/ResolveInfo.php
lib/internal/Magento/Framework/GraphQl/Config/Element/Field.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/BatchResponse.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/Value.php
lib/internal/Magento/Framework/GraphQl/Schema.php

### Fixed Issues (if relevant)

1. Fixes magento/magento2#32028

### Manual testing scenarios (*)

1. ...
2. ...

### Questions or comments

CC @sivaschenko 

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by eduard13** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Database Compare, Functional Tests B2B, Static Tests

**Comment by mrtuvn** [comment]
@magento run Static Tests, Functional Tests B2B, Database Compare

**Comment by eduard13** [comment]
@magento run Static Tests, Functional Tests B2B, Database Compare

**Review by eduard13 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @eduard13, thank you for the review.
[ENGCOM-8782](https://jira.corp.magento.com/browse/ENGCOM-8782) has been created to process this Pull Request

**Comment by eduard13** [comment]
@magento run Database Compare

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@sivaschenko interfaces mark (non-api) in my description seem got issue when add api annotation
related with problem Magento\Test\Integrity\PublicCodeTest->testAllPHPClassesReferencedFromPublicClassesArePublic()

**Comment by mrtuvn** [comment]
update + fix static
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run WebAPI Tests

**Comment by mrtuvn** [comment]
Re-targeted PR branch to 2.4-develop as internal discussed with @gabrieldagama

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Oscar** [comment]
Hi @mrtuvn !
Looks like you missed lib/internal/Magento/Framework/GraphQl/Query/Resolver/BatchServiceContractResolverInterface.php.
Please recheck.

**Comment by mrtuvn** [comment]
Nice catch! I will update

**Comment by mrtuvn** [comment]
cc: @engcom-Oscar corrected my mistakes

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Unit Tests, Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run Functional Tests EE, Functional Tests B2B

**Comment by mrtuvn** [comment]
@gabrieldagama should this PR target to branch api-changes-develop

File: lib/internal/Magento/Framework/GraphQlSchemaStitching/GraphQlReader/TypeMetaReaderInterface.php [hunk 1]
```diff
@@ -18,5 +20,5 @@ interface TypeMetaReaderInterface
      * @param \GraphQL\Type\Definition\Type $typeMeta
      * @return array|null
      */
-    public function read(\GraphQL\Type\Definition\Type $typeMeta)  : array;
+    public function read(\GraphQL\Type\Definition\Type $typeMeta): ?array;
```
Comment (gabrieldagama) [inline lib/internal/Magento/Framework/GraphQlSchemaStitching/GraphQlReader/TypeMetaReaderInterface.php]: Please, do not change the return type of this function. This is backward incompatible and may create other problems.

**Review by gabrieldagama — CHANGES_REQUESTED** [review]

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run WebAPI Tests, Unit Tests, Static Tests, Sample Data Tests EE, Sample Data Tests B2B, Functional Tests EE, Functional Tests CE, Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
cc @@gabrieldagama  updated PR branch

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Unit Tests, Static Tests, Integration Tests, Functional Tests CE, Functional Tests EE, Functional Tests B2B

**Comment by gabrieldagama** [comment]
@mrtuvn, please ignore the failures, for now, they are related to the fact that `api-changes-develop` is not up to date with `2.4-develop`, it will be fixed soon.

**Review by gabrieldagama — APPROVED** [review]

**Comment by gabrieldagama** [comment]
@magento import code to https://github.com/magento-engcom/magento2ce

**Comment by magento-engcom-team** [comment]
@gabrieldagama the branch with code successfully imported into`magento-engcom/magento2ce` repository. Branch name: `imported-magento-magento2-32101`.

**Pull request merged**